lib/resolve-package-path.js

```diff
@@ -194,11 +194,14 @@ module.exports = function resolvePackagePath(caches, name, dir) {
   // Ensure that basedir is an absolute path at this point.
   var absoluteStart = path.resolve(basedir);
 
-  // convert absoluteStart to a real path (follow links, etc).
-  absoluteStart = _getRealDirectoryPath(caches.REAL_DIRECTORY_PATH, absoluteStart);
+  while (_getRealDirectoryPath(caches.REAL_DIRECTORY_PATH, absoluteStart) === null) {
```

Comment:
Good catch! I never thought about passing a starting directory that itself didn't exist (other than null).